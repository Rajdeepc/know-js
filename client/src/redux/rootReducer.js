import { combineReducers } from "redux";
import AuthReducer from "../views/Auth/auth.reducer";
import NavbarReducer from '../components/Navbar/navbar.reducer'
import VideoPageReducer from '../views/VideoPage/video.reducer'
import HomePageReducer from '../views/Home/home.reducer'

const rootReducer =  combineReducers({
    AuthReducer,
    NavbarReducer,
    VideoPageReducer,
    HomePageReducer
})

export default rootReducer