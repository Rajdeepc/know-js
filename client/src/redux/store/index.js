import { createStore, applyMiddleware } from "redux"; // for creating store and applying middlerware like thunk
import thunk from "redux-thunk"; // for async actions
import rootReducer from "../rootReducer";
import { loadState, saveState } from "./sessionStorage";
import throttle from 'lodash/throttle';



const persistedState = loadState();

const store = createStore(
  rootReducer, 
  persistedState, 
  applyMiddleware(thunk));

store.subscribe(throttle(() => {
  saveState({
    LoginReducer: store.getState().LoginReducer
  });
},1000 ));


export default store;
