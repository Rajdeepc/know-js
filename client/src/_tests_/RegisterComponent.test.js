import React from "react";
import { render, cleanup } from "@testing-library/react";
import configureStore from "redux-mock-store";
import renderer from "react-test-renderer";
import RegisterConnectedComponent from "../components/Register/RegisterComponent";
import "@testing-library/jest-dom/extend-expect";
import { Provider } from "react-redux";
import { createStore } from "redux";
import rootReducer from "../redux/rootReducer";
import { BrowserRouter as Router } from "react-router-dom";

afterEach(cleanup);

const store = createStore(rootReducer);

describe("Connected Redux <RegisterConnectedComponent component>", () => {
  const component = renderer.create(
    <Router>
      <Provider store={store}>
        <RegisterConnectedComponent />
      </Provider>
    </Router>
  );
  test("renders RegisterConnectedComponent Component", () => {
    expect(component.toJSON()).toMatchSnapshot();
  });
});
