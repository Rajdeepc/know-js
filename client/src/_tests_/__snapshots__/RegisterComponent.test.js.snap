// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Connected Redux <RegisterConnectedComponent component> renders RegisterConnectedComponent Component 1`] = `
<form
  className=""
  noValidate={true}
  onSubmit={[Function]}
>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="formBasicEmail"
      onChange={[Function]}
      placeholder="Enter email"
      required={true}
      type="email"
    />
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 1024 1024"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M928 160H96c-17.7 0-32 14.3-32 32v640c0 17.7 14.3 32 32 32h832c17.7 0 32-14.3 32-32V192c0-17.7-14.3-32-32-32zm-40 110.8V792H136V270.8l-27.6-21.5 39.3-50.5 42.8 33.3h643.1l42.8-33.3 39.3 50.5-27.7 21.5zM833.6 232L512 482 190.4 232l-42.8-33.3-39.3 50.5 27.6 21.5 341.6 265.6a55.99 55.99 0 0 0 68.7 0L888 270.8l27.6-21.5-39.3-50.5-42.7 33.2z"
      />
    </svg>
    <div
      className="invalid-feedback"
    >
      Please provide a valid email address.
    </div>
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="formBasicUserName"
      onChange={[Function]}
      placeholder="Enter Username"
      required={true}
      type="text"
    />
    <svg
      fill="none"
      height="1em"
      stroke="currentColor"
      strokeLinecap="round"
      strokeLinejoin="round"
      strokeWidth="2"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M20 21v-2a4 4 0 0 0-4-4H8a4 4 0 0 0-4 4v2"
      />
      <circle
        cx="12"
        cy="7"
        r="4"
      />
    </svg>
    <div
      className="invalid-feedback"
    >
      Please provide a valid username.
    </div>
  </div>
  <div
    className="form-group"
  >
    <input
      className="form-control"
      id="formBasicPassword"
      onChange={[Function]}
      placeholder="Enter Password"
      required={true}
      type="password"
    />
    <svg
      fill="currentColor"
      height="1em"
      stroke="currentColor"
      strokeWidth="0"
      style={
        Object {
          "color": undefined,
        }
      }
      viewBox="0 0 24 24"
      width="1em"
      xmlns="http://www.w3.org/2000/svg"
    >
      <g>
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M18 8h2a1 1 0 0 1 1 1v12a1 1 0 0 1-1 1H4a1 1 0 0 1-1-1V9a1 1 0 0 1 1-1h2V7a6 6 0 1 1 12 0v1zM5 10v10h14V10H5zm6 4h2v2h-2v-2zm-4 0h2v2H7v-2zm8 0h2v2h-2v-2zm1-6V7a4 4 0 1 0-8 0v1h8z"
        />
      </g>
    </svg>
    <div
      className="invalid-feedback"
    >
      Please provide a valid password.
    </div>
  </div>
  <button
    className="btn btn-outline-primary"
    disabled={false}
    type="submit"
  >
    Register
  </button>
  <button
    className="btn btn-outline-danger"
    disabled={false}
    type="reset"
  >
    Reset
  </button>
</form>
`;
